src/python_testing/TC_AVSM_2_12.py

```diff
@@ -0,0 +1,294 @@
+                     "Verify that the DUT response contains a TriStateAutoEnum value. Store value as aNightVision"),
+            TestStep(12, "TH writes NightVision attribute with a new value", "DUT responds with a SUCCESS status code."),
+            TestStep(13, "TH reads NightVision attribute.", "Verify that the value is the same as was written in Step 12."),
+            TestStep(14, "TH reads Viewport attribute.",
+                     "Verify that the DUT response contains a ViewportStruct value. Store value as aViewport"),
+            TestStep(15, "TH writes Viewport attribute with a new value.", "DUT responds with a SUCCESS status code."),
+            TestStep(16, "TH reads Viewport attribute.", "Verify that the value is the same as was written in Step 15."),
+            TestStep(17, "TH reads SpeakerMuted attribute.",
+                     "Verify that the DUT response contains a bool value. Store value as aSpeakerMuted"),
+            TestStep(18, "TH writes SpeakerMuted attribute with value set to !aSpeakerMuted.",
+                     "DUT responds with a SUCCESS status code."),
+            TestStep(19, "TH reads SpeakerMuted attribute.", "Verify that the value is the same as was written in Step 18."),
+            TestStep(20, "TH reads MicrophoneMuted attribute.",
+                     "Verify that the DUT response contains a bool value. Store value as aMicrophoneMuted"),
+            TestStep(21, "TH writes MicrophoneMuted attribute with value set to !aMicrophoneMuted.",
+                     "DUT responds with a SUCCESS status code."),
+            TestStep(22, "TH reads MicrophoneMuted attribute.", "Verify that the value is the same as was written in Step 21."),
+            TestStep(23, "TH reads LocalVideoRecordingEnabled attribute.",
+                     "Verify that the DUT response contains a bool value. Store value as aLocalVideoRecordingEnabled"),
+            TestStep(24, "TH writes LocalVideoRecordingEnabled attribute with value set to !aLocalVideoRecordingEnabled.",
+                     "DUT responds with a SUCCESS status code."),
+            TestStep(25, "TH reads LocalVideoRecordingEnabled attribute.",
+                     "Verify that the value is the same as was written in Step 24."),
+        ]
+
+    @async_test_body
+    async def test_TC_AVSM_2_12(self):
+        endpoint = self.get_endpoint(default=1)
+        cluster = Clusters.CameraAvStreamManagement
+        attr = Clusters.CameraAvStreamManagement.Attributes
+
+        self.step(1)
+        # Commission DUT - already done
+
+        self.step(2)
+        if self.pics_guard(self.check_pics("AVSM.S.A000D")):
+            hdrEnabled = await self.read_single_attribute_check_success(
+                endpoint=endpoint, cluster=cluster, attribute=attr.HDRModeEnabled
+            )
+            logger.info(f"Rx'd HDRModeEnabled: {hdrEnabled}")
+
+        self.step(3)
+        if self.pics_guard(self.check_pics("AVSM.S.A000D")):
```

Comment:
Why repeat the same pics_guard check?  Just include this step within the above if